src/main/io/gps.c

```diff
@@ -1083,14 +1098,37 @@ void gpsConfigureUblox(void)
             if (gpsData.tempBaudRateIndex == 0) {
                 gpsData.tempBaudRateIndex = GPS_BAUDRATE_MAX; // slowest baud rate 9600
             } else {
-                gpsData.tempBaudRateIndex--; 
+                gpsData.tempBaudRateIndex--;
             }
             // set the FC baud rate to the new temp baud rate
             serialSetBaudRate(gpsPort, baudRates[gpsInitData[gpsData.tempBaudRateIndex].baudrateIndex]);
             initBaudRateCycleCount++;
 
             break;
-
+        case GPS_STATE_DETECT_BAUD_PASSIVE: {
+            static uint32_t lastMessages = 0;
+            if (gpsData.messages != lastMessages) { // received message since last check
+                lastMessages = gpsData.messages;
+                gpsData.state_position++;
+                if (++gpsData.state_position >= 3) { // at least 2 messages correctly received
```

Comment:
```suggestion
                gpsData.state_position += 2;
                if (gpsData.state_position >= 3) { // at least 2 messages correctly received
```

Reply:
Actually, it should increment only once. Thanks!